common_features.mk

```diff
@@ -428,8 +428,10 @@ ifeq ($(strip $(SPLIT_KEYBOARD)), yes)
         # Functions added via QUANTUM_LIB_SRC are only included in the final binary if they're called.
         # Unused functions are pruned away, which is why we can add multiple drivers here without bloat.
         ifeq ($(PLATFORM),AVR)
-            QUANTUM_LIB_SRC += i2c_master.c \
-                               i2c_slave.c
+            ifneq ($(NO_I2C),yes)
+                QUANTUM_LIB_SRC += i2c_master.c \
+                                   i2c_slave.c
+            endif
```

Comment:
Not really a fan of this change.  Especially as the way that these files are added, they should be optimized out, if they're not used.

Reply:
I don't think this diff is part of this PR. This diff is a change made in #10204 that has already been merged.

Reply:
Sorry that this is because the base is force pushed - let me rebase it first. Actually, there shouldn't a change here.